### Carga de datos
Una vez que tenemos los datos en formato csv, para poder manejarlos los cargamos anteriormene en github. Estos los importamos ahora mediante.

In [45]:
import pandas as pd
import numpy as np

url_data = 'https://raw.githubusercontent.com/dt-sc-Josue-Diaz/Proyecto-DS-II/refs/heads/main/02_data_discovery/data_para_discovery.csv'
df = pd.read_csv(url_data)


Para darnos la idea de que seguimos como en el primer paso, tenemos una muestra para observar que las configuraciones previas fueron las correctas.


In [46]:
df.sample(5)

SeriousDlqin2yrs  RevolvingUtilizationOfUnsecuredLines  age  \
134684                 0                              0.085314   47   
133329                 0                              0.793568   43   
123789                 1                              1.000000   30   
43184                  0                              0.031706   54   
25409                  0                              0.043472   53   

        NumberOfTime30-59DaysPastDueNotWorse    DebtRatio  MonthlyIncome  \
134684                                     0     0.509957        15416.0   
133329                                     2  6784.000000            0.0   
123789                                     0    27.000000            NaN   
43184                                      0     0.257097         6833.0   
25409                                      0     0.459142         5200.0   

        NumberOfOpenCreditLinesAndLoans  NumberOfTimes90DaysLate  \
134684                               18                        0   
133329                               14                        0   
123789                                1                        0   
43184                                 5                        0   
25409                                25                        0   

        NumberRealEstateLoansOrLines  NumberOfTime60-89DaysPastDueNotWorse  \
134684                             2                                     0   
133329                             3                                     0   
123789                             0                                     0   
43184                              1                                     0   
25409                              3                                     0   

        NumberOfDependents  
134684                 3.0  
133329                 1.0  
123789                 0.0  
43184                  0.0  
25409                  0.0

# Valores nulos
Vamos a proceder con los valores nulos, necesitamos conocer el timpo de variables con datos nulos, a manera de sospecha, la falta de informacion siempre es causa de riesgo a la institución y a los procedimientos de concesion de créditos.



In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150000 entries, 0 to 149999
Data columns (total 11 columns):
 #   Column                                Non-Null Count   Dtype  
---  ------                                --------------   -----  
 0   SeriousDlqin2yrs                      150000 non-null  int64  
 1   RevolvingUtilizationOfUnsecuredLines  150000 non-null  float64
 2   age                                   150000 non-null  int64  
 3   NumberOfTime30-59DaysPastDueNotWorse  150000 non-null  int64  
 4   DebtRatio                             150000 non-null  float64
 5   MonthlyIncome                         120269 non-null  float64
 6   NumberOfOpenCreditLinesAndLoans       150000 non-null  int64  
 7   NumberOfTimes90DaysLate               150000 non-null  int64  
 8   NumberRealEstateLoansOrLines          150000 non-null  int64  
 9   NumberOfTime60-89DaysPastDueNotWorse  150000 non-null  int64  
 10  NumberOfDependents                    146076 non-null  float64
dtype

De manera superficial notamos que si existen nulos, vemos ahora que la cantidad de datos es la que tenemos en la documentacion de la pagina de kaggle.

In [48]:
df.shape

(150000, 11)

Vamos a testear la cantidad porncentual de datos nulos.

In [49]:
(df.isnull().sum()/df.shape[0])*100

SeriousDlqin2yrs                         0.000000
RevolvingUtilizationOfUnsecuredLines     0.000000
age                                      0.000000
NumberOfTime30-59DaysPastDueNotWorse     0.000000
DebtRatio                                0.000000
MonthlyIncome                           19.820667
NumberOfOpenCreditLinesAndLoans          0.000000
NumberOfTimes90DaysLate                  0.000000
NumberRealEstateLoansOrLines             0.000000
NumberOfTime60-89DaysPastDueNotWorse     0.000000
NumberOfDependents                       2.616000
dtype: float64

Notemos que tenemos una cantidad de nulos cercanas al 20%, pero como pista, la falta de información es causal de riesgo a la contraperte, por tal buscamos una manera de evitar eliminar tantos registros.

In [50]:
df[df['MonthlyIncome'].isna()].describe()

SeriousDlqin2yrs  RevolvingUtilizationOfUnsecuredLines           age  \
count      29731.000000                          29731.000000  29731.000000   
mean           0.056137                              6.649421     56.362349   
std            0.230189                            217.814854     15.438786   
min            0.000000                              0.000000     21.000000   
25%            0.000000                              0.016027     46.000000   
50%            0.000000                              0.081697     57.000000   
75%            0.000000                              0.440549     67.000000   
max            1.000000                          22198.000000    109.000000   

       NumberOfTime30-59DaysPastDueNotWorse      DebtRatio  MonthlyIncome  \
count                          29731.000000   29731.000000            0.0   
mean                               0.579866    1673.396556            NaN   
std                                6.255361    4248.372895            NaN   
min                                0.000000       0.000000            NaN   
25%                                0.000000     123.000000            NaN   
50%                                0.000000    1159.000000            NaN   
75%                                0.000000    2382.000000            NaN   
max                               98.000000  329664.000000            NaN   

       NumberOfOpenCreditLinesAndLoans  NumberOfTimes90DaysLate  \
count                     29731.000000             29731.000000   
mean                          7.216071                 0.484612   
std                           4.842720                 6.250408   
min                           0.000000                 0.000000   
25%                           4.000000                 0.000000   
50%                           6.000000                 0.000000   
75%                          10.000000                 0.000000   
max                          45.000000                98.000000   

       NumberRealEstateLoansOrLines  NumberOfTime60-89DaysPastDueNotWorse  \
count                  29731.000000                          29731.000000   
mean                       0.871481                              0.452995   
std                        1.034291                              6.242076   
min                        0.000000                              0.000000   
25%                        0.000000                              0.000000   
50%                        1.000000                              0.000000   
75%                        1.000000                              0.000000   
max                       23.000000                             98.000000   

       NumberOfDependents  
count        25807.000000  
mean             0.316310  
std              0.809944  
min              0.000000  
25%              0.000000  
50%              0.000000  
75%              0.000000  
max              9.000000

En contexto con los metadatos, observamos que DebRatio, es la proporción;

$$DebRatio=\frac{Gastos}{Ingresos}$$

se mide qué proporción del ingreso mensual de una persona se destina a cubrir deudas y gastos esenciales es decir, la capacidad que tiene una persona para solverntarse además, se puede entender como un indicador de la capacidad de endeudamiento de un individuo. Notemos que DebRatio mayor que uno nos está implicando que la persona no gana lo suficiente respecto a sus gastos o no está ganando dinero. De esta manera, una persona que por mas de 3 meses tenga un debratio mayor a uno tendría sentido pensar que no está generando ingresos.

In [51]:
df.loc[df.MonthlyIncome.isna() == True,'MonthlyIncome']

6        NaN
8        NaN
16       NaN
32       NaN
41       NaN
          ..
149976   NaN
149977   NaN
149984   NaN
149992   NaN
149997   NaN
Name: MonthlyIncome, Length: 29731, dtype: float64

Hicismos una investigacion no desiva para conseguir el promedio de ingresos mensuales en el año de la información, para conseguir los datos siguientes

| **Rango de edad** | **Ingreso mensual promedio (USD)** |
|-------------------|------------------------------------|
| 16-24 años        | $1,800 - $2,200                    |
| 25-34 años        | $3,200 - $4,000                    |
| 35-44 años        | $4,500 - $5,500                    |
| 45-54 años        | $4,800 - $6,000                    |
| 55-64 años        | $4,200 - $5,000                    |
| 65 años o más     | $2,000 - $3,000                    |

De esta manera tendremos una idea de cuanto posiblemente ganaba una persona por edad en el 2011. Con eso definimos la funcion para poder asignar a cada registro un aproximado de lo que podieran ganar en ese entonces por grupo de edad.

In [52]:
def ingresos_promedio(age):
    # Respecto a la edad de la persona se le asigna un ingreso aproximado de ese año.
    if age < 25:
        return 2000
    elif age < 35:
        return 4500
    elif age < 45:
        return 4800
    else:
        return 4200

Pues bien, ahora notemos que; si el deberatio es menor que uno y no tiene ingresos mensuales entonces en verdad es una perdida de información por lo que asignarle el ingreso promedio por edad no sesgaria el estudio a diferencia de usar la media del data set. Pues como puede verse en la tabla anterior la edad es un factor en la cantidad de ingresos.

Por otro lado si el DebRatio es mayor que uno tambien tiene sentido pensar que la persona no tiene ingresos para al menos cubrir sus gastos y por tanto sus deudas. De esta manera asignamos a cero este ingreso.


In [53]:
df.loc[(df['DebtRatio'] <1) & (df['MonthlyIncome'].isna()), 'MonthlyIncome'] = df.loc[(df['DebtRatio'] < 1) & (df['MonthlyIncome'].isna()), 'age'].apply(ingresos_promedio).astype(float)
df.loc[(df['DebtRatio'] >= 1) & (df['MonthlyIncome'].isna()), 'MonthlyIncome'] = 0

In [54]:
df[df.DebtRatio ==0].describe()

SeriousDlqin2yrs  RevolvingUtilizationOfUnsecuredLines          age  \
count       4113.000000                           4113.000000  4113.000000   
mean           0.082665                              0.450194    52.124970   
std            0.275408                              0.497185    21.322905   
min            0.000000                              0.000000    21.000000   
25%            0.000000                              0.000000    32.000000   
50%            0.000000                              0.000875    52.000000   
75%            0.000000                              1.000000    70.000000   
max            1.000000                              1.000000   103.000000   

       NumberOfTime30-59DaysPastDueNotWorse  DebtRatio  MonthlyIncome  \
count                           4113.000000     4113.0    4113.000000   
mean                               5.032580        0.0    4068.413080   
std                               21.507982        0.0   10195.787426   
min                                0.000000        0.0       0.000000   
25%                                0.000000        0.0    2200.000000   
50%                                0.000000        0.0    4200.000000   
75%                                0.000000        0.0    4500.000000   
max                               98.000000        0.0  629000.000000   

       NumberOfOpenCreditLinesAndLoans  NumberOfTimes90DaysLate  \
count                      4113.000000              4113.000000   
mean                          2.223195                 5.097982   
std                           2.414849                21.498040   
min                           0.000000                 0.000000   
25%                           0.000000                 0.000000   
50%                           2.000000                 0.000000   
75%                           3.000000                 0.000000   
max                          20.000000                98.000000   

       NumberRealEstateLoansOrLines  NumberOfTime60-89DaysPastDueNotWorse  \
count                        4113.0                           4113.000000   
mean                            0.0                              5.003404   
std                             0.0                             21.513965   
min                             0.0                              0.000000   
25%                             0.0                              0.000000   
50%                             0.0                              0.000000   
75%                             0.0                              0.000000   
max                             0.0                             98.000000   

       NumberOfDependents  
count         3708.000000  
mean             0.330367  
std              0.806686  
min              0.000000  
25%              0.000000  
50%              0.000000  
75%              0.000000  
max              6.000000

In [55]:
df[df.DebtRatio>1].describe()

SeriousDlqin2yrs  RevolvingUtilizationOfUnsecuredLines           age  \
count      35137.000000                          35137.000000  35137.000000   
mean           0.065202                              7.076243     55.006261   
std            0.246885                            239.510379     14.908925   
min            0.000000                              0.000000     21.000000   
25%            0.000000                              0.022458     44.000000   
50%            0.000000                              0.105250     55.000000   
75%            0.000000                              0.490363     65.000000   
max            1.000000                          22198.000000    109.000000   

       NumberOfTime30-59DaysPastDueNotWorse      DebtRatio  MonthlyIncome  \
count                          35137.000000   35137.000000   35137.000000   
mean                               0.299286    1505.989566     465.009022   
std                                2.841582    3999.026847    1576.086349   
min                                0.000000       1.000500       0.000000   
25%                                0.000000      42.000000       0.000000   
50%                                0.000000     907.000000       0.000000   
75%                                0.000000    2210.000000       0.000000   
max                               98.000000  329664.000000   70000.000000   

       NumberOfOpenCreditLinesAndLoans  NumberOfTimes90DaysLate  \
count                     35137.000000             35137.000000   
mean                          8.071662                 0.165979   
std                           5.098427                 2.806182   
min                           0.000000                 0.000000   
25%                           4.000000                 0.000000   
50%                           7.000000                 0.000000   
75%                          11.000000                 0.000000   
max                          58.000000                98.000000   

       NumberRealEstateLoansOrLines  NumberOfTime60-89DaysPastDueNotWorse  \
count                  35137.000000                          35137.000000   
mean                       1.082933                              0.140393   
std                        1.277702                              2.784314   
min                        0.000000                              0.000000   
25%                        0.000000                              0.000000   
50%                        1.000000                              0.000000   
75%                        2.000000                              0.000000   
max                       54.000000                             98.000000   

       NumberOfDependents  
count        31689.000000  
mean             0.436650  
std              0.927896  
min              0.000000  
25%              0.000000  
50%              0.000000  
75%              0.000000  
max             10.000000

In [56]:
35137/df.shape[0]

0.23424666666666666

In [57]:
df[df.MonthlyIncome.isna()].sum()

SeriousDlqin2yrs                        0.0
RevolvingUtilizationOfUnsecuredLines    0.0
age                                     0.0
NumberOfTime30-59DaysPastDueNotWorse    0.0
DebtRatio                               0.0
MonthlyIncome                           0.0
NumberOfOpenCreditLinesAndLoans         0.0
NumberOfTimes90DaysLate                 0.0
NumberRealEstateLoansOrLines            0.0
NumberOfTime60-89DaysPastDueNotWorse    0.0
NumberOfDependents                      0.0
dtype: float64

Estos son los casos minimos en los cuales podemos hacer cambios sin ser drásticos en los datos. Además observemos que;

In [58]:
(df.isnull().sum()/df.shape[0])*100


SeriousDlqin2yrs                        0.000
RevolvingUtilizationOfUnsecuredLines    0.000
age                                     0.000
NumberOfTime30-59DaysPastDueNotWorse    0.000
DebtRatio                               0.000
MonthlyIncome                           0.000
NumberOfOpenCreditLinesAndLoans         0.000
NumberOfTimes90DaysLate                 0.000
NumberRealEstateLoansOrLines            0.000
NumberOfTime60-89DaysPastDueNotWorse    0.000
NumberOfDependents                      2.616
dtype: float64

Ahora tenemos una pérdida de información del 2.616% Si no eliminamos estos datos. Se tiene en la variable target.

In [59]:
df['SeriousDlqin2yrs'].unique()

array([1, 0])

Porcentualmente tenemos;

In [60]:
df['SeriousDlqin2yrs'].value_counts()/df.shape[0]

SeriousDlqin2yrs
0    0.93316
1    0.06684
Name: count, dtype: float64

Si eliminamos los datos que son NA, esto cambia a;

In [61]:
df = df[~df.NumberOfDependents.isna()]
df.reset_index(drop=True, inplace=True)

Lo cual nos permite no disminuir mas el objetivo de los impagos.

In [62]:
(df.isnull().sum()/df.shape[0])*100

SeriousDlqin2yrs                        0.0
RevolvingUtilizationOfUnsecuredLines    0.0
age                                     0.0
NumberOfTime30-59DaysPastDueNotWorse    0.0
DebtRatio                               0.0
MonthlyIncome                           0.0
NumberOfOpenCreditLinesAndLoans         0.0
NumberOfTimes90DaysLate                 0.0
NumberRealEstateLoansOrLines            0.0
NumberOfTime60-89DaysPastDueNotWorse    0.0
NumberOfDependents                      0.0
dtype: float64

In [63]:
df.NumberOfDependents.describe()

count    146076.000000
mean          0.757222
std           1.115086
min           0.000000
25%           0.000000
50%           0.000000
75%           1.000000
max          20.000000
Name: NumberOfDependents, dtype: float64

En este contexto creemos que no podemos tratar de imputar estos nulos, ya que la cantidad de personas por familia impactan mucho en el debratio. Por lo que agreagar datos podria sesgar de manera intencional un registro a una clase. Preferimos eliminarlos además que ahora solo es el 	2.616 porciento del total de la muestra y el impacto de la variable objetivo consideramos que es tolerable, tendremos esto en cuenta al momento del entrenamiento.

# Notas importantes

In [64]:
df.NumberOfDependents.mean()

np.float64(0.7572222678605657)

In [65]:
df['NumberOfDependents'].value_counts()/df.shape[0]

NumberOfDependents
0.0     0.594909
1.0     0.180153
2.0     0.133643
3.0     0.064918
4.0     0.019593
5.0     0.005107
6.0     0.001082
7.0     0.000349
8.0     0.000164
10.0    0.000034
9.0     0.000034
20.0    0.000007
13.0    0.000007
Name: count, dtype: float64

In [66]:
sum(df.loc[df.SeriousDlqin2yrs==1,'age'] <= 25)

311

In [67]:
sum(df.loc[df.SeriousDlqin2yrs==1,'age'] >= 75)

218

### Duplicados

In [68]:
# A nivel general
df.duplicated().sum()

np.int64(520)

In [69]:
# A nivel PK
df.index.nunique()==df.shape[0]

True

## Nulos
Esta parte evalua si existen datos nulos.

In [70]:
df.isnull().sum()

SeriousDlqin2yrs                        0
RevolvingUtilizationOfUnsecuredLines    0
age                                     0
NumberOfTime30-59DaysPastDueNotWorse    0
DebtRatio                               0
MonthlyIncome                           0
NumberOfOpenCreditLinesAndLoans         0
NumberOfTimes90DaysLate                 0
NumberRealEstateLoansOrLines            0
NumberOfTime60-89DaysPastDueNotWorse    0
NumberOfDependents                      0
dtype: int64

No se observan datos nulos. Con eso se finaliza la integridad de los datos para el eda.

# Metadatos
Tenemos valores unicos en nuestro data set. Las siguientes lineas nos permitirámn hacer bien la estadística en la siguiente sección.

In [71]:
df.columns

Index(['SeriousDlqin2yrs', 'RevolvingUtilizationOfUnsecuredLines', 'age',
       'NumberOfTime30-59DaysPastDueNotWorse', 'DebtRatio', 'MonthlyIncome',
       'NumberOfOpenCreditLinesAndLoans', 'NumberOfTimes90DaysLate',
       'NumberRealEstateLoansOrLines', 'NumberOfTime60-89DaysPastDueNotWorse',
       'NumberOfDependents'],
      dtype='object')

## Diccionario de variales.

| Variable Name                                  | Description                                                                                         | Type      |
|------------------------------------------------|-----------------------------------------------------------------------------------------------------|-----------|
| SeriousDlqin2yrs                               | Person experienced 90 days past due delinquency or worse                                             | Y/N       |
| RevolvingUtilizationOfUnsecuredLines           | Total balance on credit cards and personal lines of credit except real estate divided by credit limits| percentage|
| age                                            | Age of borrower in years                                                                             | integer   |
| NumberOfTime30-59DaysPastDueNotWorse           | Number of times borrower has been 30-59 days past due but no worse in the last 2 years               | integer   |
| DebtRatio                                      | Monthly debt payments, alimony, living costs divided by monthly gross income                         | percentage|
| MonthlyIncome                                  | Monthly income                                                                                       | real      |
| NumberOfOpenCreditLinesAndLoans                | Number of open loans (installment like car loan or mortgage) and lines of credit (e.g. credit cards) | integer   |
| NumberOfTimes90DaysLate                        | Number of times borrower has been 90 days or more past due                                           | integer   |
| NumberRealEstateLoansOrLines                   | Number of mortgage and real estate loans including home equity lines of credit                       | integer   |
| NumberOfTime60-89DaysPastDueNotWorse           | Number of times borrower has been 60-89 days past due but no worse in the last 2 years               | integer   |
| NumberOfDependents                             | Number of dependents in family excluding themselves (spouse, children, etc.)                         | integer   |

Vamos a contextualizar la tabla anterior.


| Nombre de la variable| Descripción | Tipo|
|------------------------------------------------|-----------------------------------------------------------------------------------------------------|-----------|
| moroso |  Persona que no ha pagado en mas de 90 dias| Moroso/ No moroso  |
| Porcentaje crediticio | Proporcion de la linea de credito usada| Porcentaje|
|edad| Edad del cliente en años | entero |
|Retrasos bimestrales | Cantidad de veces que no se ha pagado en dos meses en una plazo de dos años | entero   |
| razon_de_deuda| Deudas dividido por el ingreso bruto | porcerntaje|
| ingreso_mensual| Ingreso mensual | decimal |
| cantidad_de_prestamos | Número de préstamos abiertos (a plazos, como préstamos para automóviles o hipotecas) y líneas de crédito (por ejemplo, tarjetas de crédito)| entero   |
|cartera_vencida| Número de veces que el prestatario ha estado vencido por 90 días o más| entero   |
| deuda_inmobiliara |Número de préstamos hipotecarios y inmobiliarios, incluidas líneas de crédito sobre el valor líquido de la vivienda| entero   |
| retrasos_trimestrales| Número de veces que el prestatario ha estado vencido entre 60 y 89 días pero no ha empeorado en los últimos 2 años| entero   |
|numero_dependientes | Cantidad de personas en la familia | entero   |

Por tanto vamos a renombrar para nuestro estudio.


In [72]:
df.columns = ['moroso', 'porcentaje_crediticio', 'edad', 'retrasos_bimestrales', 'razon_de_deuda','ingresos', 'cantidad_de_prestamos', 'cartera_vencida','deuda_inmobiliara','retrasos_trimestrales','numero_dependientes']

In [73]:
def categoria_edades(edad):
    # Aquí debes retornar un solo valor basado en la edad, por ejemplo:
    if edad < 35:
        return 'Joven'
    elif edad < 45:
        return 'Adulto Joven'
    elif edad < 65:
        return 'Adulto'
    else:
        return 'Adulto mayor'

Q1 = df['ingresos'].quantile(0.25)
Q2 = df['ingresos'].quantile(0.75)

# Esta fución depende totalmente de los datos. No pudimos conseguir informacion sobre esto, por lo que pensamos que es la mejor manera de segregar la información.
def clases_social(ingreso):
  if ingreso < Q1:
    return 'Bajo'
  elif ingreso < Q2:
    return 'Medio'
  else:
    return 'Alto'

def categoria(variable):
    # Respecto a la edad de la persona se le asigna un ingreso aproximado de ese año.
    if variable == 1 :
        return "Moroso"
    else:
        return "No moroso"

In [74]:
df['ingresos'].quantile(0.25),  df['ingresos'].quantile(0.75)

(np.float64(2000.0), np.float64(7500.0))

In [75]:
df['generacion'] = df.edad.apply(categoria_edades)
df['clase_social'] = df.ingresos.apply(clases_social)
df['moroso'] = df['moroso'].apply(categoria)
df['numero_dependientes'] = df['numero_dependientes'].astype(int)

In [76]:
df

moroso  porcentaje_crediticio  edad  retrasos_bimestrales  \
0          Moroso               0.766127    45                     2   
1       No moroso               0.957151    40                     0   
2       No moroso               0.658180    38                     1   
3       No moroso               0.233810    30                     0   
4       No moroso               0.907239    49                     1   
...           ...                    ...   ...                   ...   
146071  No moroso               0.040674    74                     0   
146072  No moroso               0.299745    44                     0   
146073  No moroso               0.246044    58                     0   
146074  No moroso               0.000000    30                     0   
146075  No moroso               0.850283    64                     0   

        razon_de_deuda  ingresos  cantidad_de_prestamos  cartera_vencida  \
0             0.802982    9120.0                     13                0   
1             0.121876    2600.0                      4                0   
2             0.085113    3042.0                      2                1   
3             0.036050    3300.0                      5                0   
4             0.024926   63588.0                      7                0   
...                ...       ...                    ...              ...   
146071        0.225131    2100.0                      4                0   
146072        0.716562    5584.0                      4                0   
146073     3870.000000       0.0                     18                0   
146074        0.000000    5716.0                      4                0   
146075        0.249908    8158.0                      8                0   

        deuda_inmobiliara  retrasos_trimestrales  numero_dependientes  \
0                       6                      0                    2   
1                       0                      0                    1   
2                       0                      0                    0   
3                       0                      0                    0   
4                       1                      0                    0   
...                   ...                    ...                  ...   
146071                  1                      0                    0   
146072                  1                      0                    2   
146073                  1                      0                    0   
146074                  0                      0                    0   
146075                  2                      0                    0   

          generacion clase_social  
0             Adulto         Alto  
1       Adulto Joven        Medio  
2       Adulto Joven        Medio  
3              Joven        Medio  
4             Adulto         Alto  
...              ...          ...  
146071  Adulto mayor        Medio  
146072  Adulto Joven        Medio  
146073        Adulto         Bajo  
146074         Joven        Medio  
146075        Adulto         Alto  

[146076 rows x 13 columns]

In [77]:
df.to_csv('03_eda/data_para_eda.csv',index = False)